Sessions/5b-word2vec-training.ipynb

```diff
@@ -0,0 +1,742 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "EEjIZfUkzA0f"
+   },
+   "source": [
+    "# Word2Vec: training your own model"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook we will see how you can train your own Word2Vec model.\n",
+    "\n",
+    "There are two main steps:\n",
+    "* Preprocessing the corpus you will train your model on.\n",
+    "* Training the model.\n",
+    "\n",
+    "Let's go!"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "yePUXhVizA0g"
+   },
+   "source": [
+    "## Preprocessing"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "kCsjMvZCzA0h"
+   },
+   "source": [
+    "Word2Vec requires the training data to be provided as an iterable (e.g. a list of tokens)."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define your preprocessing"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Many different choices can be made when preprocessing a corpus to be used as Word2Vec training data. Ultimately the choices you'll make will usually depend on:\n",
+    "\n",
+    "1. The size of your corpus (it might be computationally too expensive to preprocess very large corpora: choose a quality-time trade-off wisely!)\n",
+    "2. What you are planning to use the model for (e.g. maybe you want to look at non-lemmatized tokens? Maybe you'd like to keep numbers?)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For this first complete example, our preprocessing will consist of:\n",
+    "- Lowercasing\n",
+    "- Punctuation removal\n",
+    "- Stopword removal\n",
+    "- Lemmatization\n",
+    "\n",
+    "But preprocessing may also include tasks like:\n",
+    "- Removal of tokens with less than _n_ characters\n",
+    "- Removal of numbers\n",
+    "- PoS tagging"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we are experts in spacy now, we will use it for preprocessing our corpus."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "_EogcuS4zA0h"
+   },
+   "outputs": [],
+   "source": [
+    "import spacy"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "f_8LBHHozA0i"
+   },
+   "source": [
+    "Load the English model `en_core_web_sm` from `spacy`:"
```

Comment:
Show how to install this model? This may fail on colab/binder? or if someone has not run the previous notebooks properly?

Reply:
e.g. !python3 -m spacy download en_core_web_sm